cpp/src/slice2cs/Gen.cpp

```diff
@@ -3117,81 +3084,38 @@ Slice::Gen::TypesVisitor::visitDataMember(const DataMemberPtr& p)
     emitDeprecate(p, cont, _out, "member");
 
     string type = typeToString(p->type(), ns, isOptional);
-    string propertyName = fixId(p->name(), baseTypes, isClass);
-    string dataMemberName;
+    string dataMemberName = fixId(p->name(), baseTypes, isClass);
 
-    if(isProperty)
-    {
-        dataMemberName = "_" + p->name();
-    }
-    else
+    emitAttributes(p);
+    emitGeneratedCodeAttribute();
+    if(isPrivate)
```

Comment:
The PR didn't change the logic, but I don't see the point of generating private properties. I think we should consider just dropping this "protected" attribute at least for structs.

Reply:
I agree, we should fix this code further in a follow-up PR.

The virtual on "mapped struct" properties is also totally useless.